module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -634,7 +636,25 @@ public function getThumbnail($size = 'small')
                 }
             }
         }
-        return $closestMatch;
+
+        // If EDS actually returned cover image data, use it.  EDS only provides this data
+        // for certain ebook packages.
+        if ($closestMatch) {
+            return $closestMatch;
+        }
+
+        // Optionally use VuFind's default cover loader
+        $fallBackToCoverLoader = $this->recordConfig?->Cover?->fallBackToCoverLoader?->toArray() ?? [];
+        if ($fallBackToCoverLoader) {
+            $parentThumbnail = parent::getThumbnail($size);
+
+            // Only use the default cover loader if the record contained at least one configured field
+            if (array_intersect_key($parentThumbnail, array_flip($fallBackToCoverLoader))) {
```

Comment:
Purely a matter of taste, but this achieves the same effect in a slightly shorter (and to my brain, easier to understand) line:

```suggestion
            if (array_intersect(array_keys($parentThumbnail), $fallBackToCoverLoader)) {
```

...is there a reason the other way is better?

Reply:
I think there is a slight performance advantage for using `array_intersect_key` over `array_intersect` and `array_keys`. That said, these are not large arrays so I doubt we'd notice any real difference.

Reply:
I wasn't sure how `array_flip` performance compares to `array_keys` -- that may be a minor factor too. In any case, as you say, I don't think it makes a huge difference one way or another and I'm happy to defer to @maccabeelevine's preference.

Reply:
I took the suggested change, I agree it's easier to understand.  The original version hurt my brain a bit even when I wrote it.  Agree with @meganschanz that for arrays this tiny the performance difference is less important.